File: leetcode/2214 Minimum Health to Beat Game/solution.go
func minimumHealth(damage []int, armor int) int64 { // 贪心抵消max或armor的伤害！！！
    sum, max := 0, 0
    for _, current := range damage {
        sum += current
        max = maxTwo(max, current)
    }
    return int64(sum - minTwo(max, armor) + 1) // 存活需要剩一滴血！！！
}

func maxTwo(a int, b int) int {
    if a >= b {
        return a
    } else {
        return b
    }
}

func minTwo(a int, b int) int {
    if a <= b {
        return a
    } else {
        return b
    }
}

File: leetcode/2214 Minimum Health to Beat Game/solution.java
class Solution {
    public long minimumHealth(int[] damage, int armor) { // 贪心抵消max或armor的伤害！！！
        long max = 0, sum = 0;
        for (int current : damage) {
            sum += current;
            max = Math.max(max, current);
        }
        return (long) sum - Math.min(max, armor) + 1; // 存活需要剩一滴血！！！
    }
}